src/__tests__/utils/puzzles.test.ts

```diff
@@ -0,0 +1,22 @@
+import { checkPuzzleAnswer } from '../../utils/puzzles';
+import { mockPuzzle } from '../../__mocks__/pages/api/puzzles';
+import { mockRandomSeed } from '../../__mocks__/pages/api/puzzles/instances/submit.ts';
+import {getRandomNumber, getRandomString} from '../../__mocks__/pages/api/getRandom';
+import puzzle from "../../__mocks__/mock_puzzle";
+
+it('should compare the user submitted puzzle answer with the ans key in the puzzle (from the db)', () => {
+  let pzl;
+  let ans;
+  try{
+    pzl = pzl = mockPuzzle();
+    ans = mockPuzzle().variables?.answer;
```

Comment:
check if the mockPuzzle has variables, else, update the mockPuzzle to have variables

Reply:
Nah I think we should just have a mock for MCQ in addition to a mock for text answer

Reply:
1. mockPuzzle is updated, so it does have "variables"
2. I kept the static mock puzzle in case the mockPuzzle fail to have values in variable or answer